app/services/users/auth_checker.rb

```diff
@@ -21,7 +21,7 @@ def authenticated?
 
       if @user.authenticated_locally?
         @user.valid_password?(@password)
-      elsif LdapAuthentication.configured?
+      elsif defined?(LdapAuthentication) && LdapAuthentication.configured?
```

Comment:
Instead of resorting to `defined?`, should `uses_ldap_authentication` maybe become a configuration setting on `Rails.application.config` that defaults to `false`, and we set to `true` only for NU? It seems like it would be easier to manage in tests, too (we can change that boolean at will to exercise the code path we’re interested in, without any fake classes just to have a constant present).

Reply:
I like the idea, just moved the configuration to `settings.yml`.